libcodechecker/analyze/log_parser.py

```diff
@@ -93,14 +93,28 @@ def parse_compile_commands_json(logfile, add_compiler_defaults=False):
     counter = 0
     for entry in data:
         sourcefile = entry['file']
+
+        if not os.path.isabs(sourcefile):
+            # Newest versions of intercept-build can create the 'file' in the
+            # JSON Compilation Database as a relative path.
+            sourcefile = os.path.join(os.path.abspath(entry['directory']),
+                                      sourcefile)
+
         lang = option_parser.get_language(sourcefile[sourcefile.rfind('.'):])
 
         if not lang:
             continue
 
         action = build_action.BuildAction(counter)
 
-        command = entry['command']
+        if 'command' in entry:
+            command = entry['command']
+        elif 'arguments' in entry:
+            # Newest versions of intercept-build create an argument vector
+            # instead of a command string.
+            command = ' '.join(entry['arguments'])
```

Comment:
Not sure that i get the question... the point i was trying to rise was that when you read the compilation database, it's better to keep `arguments` as list (don't concatenate it) and if it was a `command` make it as a list (with `shlex.split` or something clever)... So either way, you get a list of arguments... So, you can call `subprocess.call` or `subprocess.check_call` without `shell=True`. (Which will make your code more portable and get rid of the problem i've mentioned before.)